Polymorphic Types/Covariance/src/Covariance.scala

```diff
@@ -1,9 +1,10 @@
 object Covariance {
   class Animal[+T](val animal: T)
 
-  class Reptile extends Animal
+  trait AnimalKind
```

Comment:
I am really surprised that this code compiles with Scala 2… Thank you for fixing it!

However, to be honest the code in this file does not make a lot of sense to me, and I don’t see it in the page https://www.scala-exercises.org/scala_tutorial/polymorphic_types. Is it actually used somewhere?